src/stores/orderStore.ts

```diff
@@ -51,92 +52,101 @@ class OrderStore extends BaseStore {
     switch (orderType) {
       case OrderType.Market:
-        return this.api
-          .placeMarket(body)
-          .then(this.orderPlacedSuccessfully, this.orderPlacedUnsuccessfully)
-          .then(() => Promise.resolve());
+        return (
+          this.api
+            .placeMarket(body)
+            // tslint:disable-next-line:no-empty
+            .then(() => {}, this.orderPlacedUnsuccessfully)
+            .then(() => Promise.resolve())
+        );
       case OrderType.Limit:
-        return this.api
-          .placeLimit(body)
-          .then(this.orderPlacedSuccessfully, this.orderPlacedUnsuccessfully)
-          .then(() => Promise.resolve());
+        return (
+          this.api
+            .placeLimit(body)
+            // tslint:disable-next-line:no-empty
+            .then(() => {}, this.orderPlacedUnsuccessfully)
+            .then(() => Promise.resolve())
+        );
     }
   };
 
   editOrder = async (body: any, id: string) =>
     this.api
       .cancelOrder(id)
       .then(() => this.api.placeLimit(body))
-      .then(this.updateOrders)
-      .then(this.orderEditedSuccessfully)
       .catch(this.orderPlacedUnsuccessfully);
 
   cancelOrder = async (id: string) => {
     await this.api.cancelOrder(id);
-    this.notificationStore.addNotification(
-      levels.information,
-      `${messages.orderCancelled} ${id}`
-    );
-    this.updateOrders();
   };
 
   cancelAll = async () => {
-    const orders = this.rootStore.orderListStore.limitOrders;
-    const promises = orders.map((order: OrderModel) => {
-      return this.api.cancelOrder(order.id);
+    this.rootStore.orderListStore.limitOrders.forEach((order: OrderModel) => {
+      this.api.cancelOrder(order.id);
     });
-    await Promise.all(promises).then(() =>
-      this.notificationStore.addNotification(
-        levels.information,
-        messages.allOrdersCanceled
-      )
-    );
-    this.updateOrders();
-  };
-
-  executeOrder = (orders: any[]) => {
-    this.rootStore.orderListStore.fetchAll().then(() => {
-      orders.forEach(order =>
-        this.rootStore.orderListStore.limitOrders.forEach(limit => {
-          const match = limit.id === order.id;
-
-          if (match) {
-            if (limit.volume === order.volume) {
-              this.notificationStore.addNotification(
-                levels.success,
-                messages.orderExecuted(order.id)
-              );
-            } else {
-              this.notificationStore.addNotification(
-                levels.success,
-                messages.orderExecutedPartially(order.id, order.volume)
-              );
-            }
-          }
-        })
-      );
+  };
 
-      this.updateOrders();
-    });
+  subscribe = (ws: any) => {
+    ws.subscribe(topics.orders, this.onOrders);
+  };
+
+  onOrders = (args: any) => {
+    const order = args[0][0];
+    switch (order.Status) {
+      case OrderBookType.Cancelled:
+        this.rootStore.orderListStore.deleteOrder(order.Id);
```

Comment:
back to my question above, i cannot see the use of _boolean_ for the `deleteOrder`